src/client/dfs/dfs.c

```diff
@@ -1075,10 +1074,11 @@ open_symlink(dfs_t *dfs, dfs_obj_t *parent, int flags, daos_oclass_id_t cid,
 		entry->value_len = value_len;
 		rc = insert_entry(parent->oh, DAOS_TX_NONE, sym->name, len,
 				  DAOS_COND_DKEY_INSERT, entry);
-		if (rc) {
+		if (rc == EEXIST) {
+			D_FREE(sym->value);
+		} else if (rc != 0) {
 			D_FREE(sym->value);
-			D_ERROR("Inserting entry %s failed (rc = %d)\n",
-				sym->name, rc);
+			D_ERROR("Inserting entry '%s' failed: %d (%s)\n", sym->name, rc, strerror(rc));
```

Comment:
(style)  line over 100 characters

Reply:
I know this warning came from me but can you fix it.

Reply:
Just fixed it locally. thx!